vespa-hadoop/src/main/java/com/yahoo/vespa/hadoop/pig/VespaDocumentOperation.java

```diff
@@ -408,45 +529,72 @@ private static boolean shouldWriteField(String name, Properties properties, int
         return true;
     }
 
-    private static void writeTensor(Map<Object, Object> map, JsonGenerator g) throws IOException {
-        g.writeFieldName("cells");
+    private static void writeTensor(Map<Object, Object> map, JsonGenerator g, Boolean isRemoveTensor) throws IOException {
```

Comment:
Can we separate removeTensor and add/create into two method since they are quite different?
Since the common part is not much, two different purpose method would be easier to read rather than a complex method with multiple functionality.